### ISSUE TITLE
Add simple AI to the opponent (right side paddle)

### ISSUE THREAD

**Issue by PhoenixFlame101** [PhoenixFlame101/pong-with-godot#1]
## The Problem

Right now both the paddles are controlled by the player(s). This issue is a feature request for a simple AI for the right paddle.

## Possible Solutions

Implementing a simple pong AI means having the right paddle move towards where the ball would end up if it kept travelling in the direction it's currently travelling in. Do not account for wall bounces since this will make the AI basically unbeatable. Recalculate where the ball will end up if it bounces against a wall.

In other words, the desired position is calculated by casting a ray from the current ball position, along its velocity vector, and seeing if it intersects the line which the AI is moving along. If it intersects a wall, the velocity changes, and tested again.

---
 - Block time: 45 mins (You need to give updates every 45 mins if you have been assigned)
 - Bounty points: 80

**Comment by tnayagarwal** [comment]
@PhoenixFlame101

**Comment by Vaarun-Kamath** [comment]
hey can i work on this pls :)
